tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,338 @@
+  """
+  image_feature = feature_dict[transformed_name(IMAGE_KEY)]
+  batch_size = tf.shape(image_feature)[0]
+  image_feature = tf.image.random_flip_left_right(image_feature)
+  image_feature = tf.image.resize_with_crop_or_pad(image_feature, 250, 250)
+  image_feature = tf.image.random_crop(image_feature, (batch_size, 224, 224, 3))
+  feature_dict[transformed_name(IMAGE_KEY)] = image_feature
+  return feature_dict
+
+# TFX Transform will call this function.
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  outputs = {}
+
+  # tf.io.decode_png function cannot be applied on a batch of data
+  image_features = tf.map_fn(lambda x: tf.io.decode_png(x[0], channels=3),
+                             inputs[IMAGE_KEY], dtype=tf.uint8)
+  image_features = tf.cast(image_features, tf.float32)
+  image_features = tf.image.resize(image_features, [224, 224])
+  image_features = tf.keras.applications.mobilenet. \
+                         preprocess_input(image_features)
+
+  outputs[transformed_name(IMAGE_KEY)] = image_features
+  # TODO(b/157064428): Support label transformation for Keras.
+  # Do not apply label transformation as it will result in wrong evaluation.
+  outputs[transformed_name(LABEL_KEY)] = inputs[LABEL_KEY]
+
+  return outputs
+
+def _write_metadata(model_path: str,
+                    label_map_path: str,
+                    mean: list,
+                    std: list):
+  """ Add TFLite metadata to the model
+
+  Args:
+    model_path: The path of the TFLite model
+    label_map_path: The path of the label map file
+    mean: The mean value used to normalize input image tensor
+    std: The standard deviation used to normalize input image tensor
+  """
+
+  model_meta = _metadata_fb.ModelMetadataT()
+
+  # Creates input normalization info.
+  input_meta = _metadata_fb.TensorMetadataT()
+  input_normalization = _metadata_fb.ProcessUnitT()
+  input_normalization.optionsType = (
+      _metadata_fb.ProcessUnitOptions.NormalizationOptions)
+  input_normalization.options = _metadata_fb.NormalizationOptionsT()
+  input_normalization.options.mean = mean
+  input_normalization.options.std = std
+  input_meta.processUnits = [input_normalization]
+
+  # Creates output label map info.
+  output_meta = _metadata_fb.TensorMetadataT()
+  label_file = _metadata_fb.AssociatedFileT()
+  label_file.name = os.path.basename(label_map_path)
+  label_file.type = _metadata_fb.AssociatedFileType.TENSOR_AXIS_LABELS
+  output_meta.associatedFiles = [label_file]
+
+  # Creates subgraph info.
+  subgraph = _metadata_fb.SubGraphMetadataT()
+  subgraph.inputTensorMetadata = [input_meta]
+  subgraph.outputTensorMetadata = [output_meta]
+  model_meta.subgraphMetadata = [subgraph]
+
+  b = flatbuffers.Builder(0)
```

Comment:
Add some inline comment on how the data is encoded with flatbuffers, for readers who are not familiar with this.

Reply:
+1 Perhaps some content here would be relevant: https://www.tensorflow.org/lite/convert/metadata